lib/postgrex.ex

```diff
@@ -454,4 +455,17 @@ defmodule Postgrex do
   defp defaults(opts) do
     Keyword.put_new(opts, :timeout, @timeout)
   end
+
+  defp ensure_deps_started!(opts) do
+    if Keyword.get(opts, :ssl, false) and Application.start(:ssl) != {:error, {:already_started, :ssl}} do
```

Comment:
`Application.start` has side effects, and depending how Postgrex is used it may still work fine in dev and not in a release. For example if Postgrex is placed under a pool it may succeed on second start attempt in dev because ssl will be started by the first attempt. However in release it would fail.

Reply:
@fishcakez can you suggest a better way to check is application loaded?

Reply:
For example `Application.loaded_applications` returns a list of loaded applications. Check `Application` and `:application` modules.

Reply:
@fishcakez That was my first idea how to check them, but loaded doesn't mean started. It is possible that you have :ssl application, but because it's not started you still would not be able to establish an SSL connection..